cstar/roms/input_dataset.py

```diff
@@ -16,18 +22,146 @@ class ROMSInputDataset(InputDataset, ABC):
+
+        Parameters:
+        -----------
+        local_dir (str or Path):
+           The directory in which to save the input dataset netCDF file
+        start_date,end_date (dt.datetime, optional):
+           If the dataset to be created is time-varying, it is made using these dates
+        np_xi, np_eta (int, optional):
+           If desired, save a partitioned copy of the input dataset to be used when
+           running ROMS in parallel. np_xi is the number of x-direction processors,
+           np_eta is the number of y-direction processors
+
+        """
+
+        # If it's not a yaml, we're done
+        if self.source.source_type != "yaml":
+            raise ValueError(
+                "Attempted to call `ROMSInputDataset.get_from_yaml() "
+                + "but ROMSInputDataset.source.source_type is "
+                + f"{self.source.source_type}, not 'yaml'"
+            )
+
+        # Ensure we're working with a Path object
+        local_dir = Path(local_dir)
+
+        # First, get the file as usual
+        self.get(local_dir)
+
+        # Make sure that the local copy is not a symlink
+        # (as InputDataset.get() symlinks files that didn't need to be downloaded)
+        yaml_file = local_dir / Path(self.source.location).name
+        if yaml_file.is_symlink():
+            actual_path = yaml_file.resolve()
+            yaml_file.unlink()
+            shutil.copy2(actual_path, yaml_file)
+            yaml_file = actual_path
+
+        # Now modify the local copy of the yaml file as needed:
+        with open(yaml_file, "r") as F:
+            _, header, yaml_data = F.read().split("---", 2)
+            yaml_dict = yaml.safe_load(yaml_data)
+
+        yaml_keys = list(yaml_dict.keys())
+        if len(yaml_keys) == 1:
+            roms_tools_class_name = yaml_keys[0]
+        elif len(yaml_keys) == 2:
+            roms_tools_class_name = [y for y in yaml_keys if y != "Grid"][0]
+        else:
+            raise ValueError(
+                f"roms tools yaml file has {len(yaml_keys)} sections. "
+                + "Expected 'Grid' and one other class"
+            )
+
+        start_time = start_date.isoformat() if start_date is not None else None
+        end_time = end_date.isoformat() if end_date is not None else None
+
+        yaml_entries_to_modify = {
+            "start_time": start_time,
+            "ini_time": start_time,
+            "end_time": end_time,
+        }
+
+        for key, value in yaml_entries_to_modify.items():
+            if key in yaml_dict[roms_tools_class_name].keys():
+                yaml_dict[roms_tools_class_name][key] = value
+
+        with open(yaml_file, "w") as F:
+            F.write(f"---{header}---\n" + yaml.dump(yaml_dict))
+
+        # Finally, make a roms-tools object from the modified yaml
+        import roms_tools
+
+        roms_tools_class = getattr(roms_tools, roms_tools_class_name)
+        roms_tools_class_instance = roms_tools_class.from_yaml(yaml_file)
```

Comment:
This is what I want to see removed to solve #84. (But it doesn't need to be done in this PR)